packages/venia-ui/lib/components/Header/cartTrigger.js

```diff
@@ -32,15 +39,29 @@ const CartTrigger = props => {
         <span className={classes.counter}>{itemCount}</span>
     ) : null;
 
+    // Because this button behaves differently on desktop and mobile
+    // we render two buttons that differ only in their click handler
+    // and control which one displays in CSS.
     return (
-        <button
-            aria-label={buttonAriaLabel}
-            className={classes.root}
-            onClick={handleClick}
-        >
-            <Icon classes={iconClasses} src={ShoppingCartIcon} />
-            {itemCounter}
-        </button>
+        <Fragment>
+            <button
+                aria-label={buttonAriaLabel}
+                className={classes.root_desktop}
+                onClick={handleDesktopClick}
+            >
+                <Icon classes={iconClasses} src={ShoppingCartIcon} />
+                {itemCounter}
+            </button>
+            <button
+                aria-label={buttonAriaLabel}
+                className={classes.root_mobile}
+                onClick={handleMobileClick}
+            >
+                <Icon classes={iconClasses} src={ShoppingCartIcon} />
+                {itemCounter}
+            </button>
+            <MiniCart isOpen={miniCartIsOpen} setIsOpen={setMiniCartIsOpen} />
```

Comment:
So instead of a drawer the mini cart is now a child of the trigger. Should make anchoring the mini cart element to the trigger a bit easier.